Sources/NIO/ByteBuffer-views.swift

```diff
@@ -187,3 +187,32 @@ extension ByteBuffer {
         self = view._buffer.getSlice(at: view.startIndex, length: view.count)!
     }
 }
+
+extension ByteBufferView: Equatable {
+    /// required by `Equatable`
+    public static func == (lhs: ByteBufferView, rhs: ByteBufferView) -> Bool {
+
+        guard lhs._range.count == rhs._range.count else {
+            return false 
+        }
+
+        let leftBufferSlice = lhs._buffer.getSlice(at: lhs._range.startIndex, length: lhs._range.count)
+        let rightBufferSlice = rhs._buffer.getSlice(at: rhs._range.startIndex, length: rhs._range.count)
+        
+        return leftBufferSlice == rightBufferSlice
+    }
+}
+
+extension ByteBufferView: Hashable {
+    /// required by `Hashable`
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(_buffer.getSlice(at: _range.startIndex, length: _range.count))
```

Comment:
Similarly, we should force-unwrap the call to `getSlice` here, with the same commented reasoning as above.

Also, a minor nit: SwiftNIO style is to explicitly use the `self` keyword to access member fields and functions.